pinot-core/src/main/java/org/apache/pinot/core/plan/TransformPlanNode.java

```diff
@@ -76,7 +83,19 @@ private void extractColumnsAndTransforms(@Nonnull BrokerRequest brokerRequest,
       IndexSegment indexSegment) {
     if (brokerRequest.isSetAggregationsInfo()) {
       for (AggregationInfo aggregationInfo : brokerRequest.getAggregationsInfo()) {
-        if (!aggregationInfo.getAggregationType().equalsIgnoreCase(AggregationFunctionType.COUNT.getName())) {
+        if (aggregationInfo.getAggregationType().equalsIgnoreCase(AggregationFunctionType.DISTINCT.getName())) {
+          // handle DISTINCT (col1, col2 ...) as an aggregate function
+          final String multiColumnExpression = AggregationFunctionUtils.getColumn(aggregationInfo);
+          final String[] distinctColumnExpressions = multiColumnExpression.split(FunctionCallAstNode.DISTINCT_MULTI_COLUMN_SEPARATOR);
+          int col = 0;
+          for (String distinctColumnExpr : distinctColumnExpressions) {
+            TransformExpressionTree transformExpressionTree = TransformExpressionTree.compileToExpressionTree(distinctColumnExpr);
+            transformExpressionTree.getColumns(_projectionColumns);
+            _expressionTrees.add(transformExpressionTree);
+            _projectionColumnIndices.put(distinctColumnExpr, col++);
```

Comment:
Where/how is this being used?

Reply:
This is being used in the ProjectionOperator to correctly store info on each projected column type and the type info is used later when we build DataTable in InstanceResponeBlock to correctly set the type in DataSchema for each projected column in distinct query